sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -467,28 +467,55 @@ class Column(protected[sql] val expr: Expression) extends Logging {
   }
 
   /**
+   * @group expr_ops
+   * @deprecated As of 1.6.0, replaced by `isnan`. This will be removed in Spark 2.0.
+   */
+  @deprecated("Use isnan. This will be removed in Spark 2.0.", "1.6.0")
+  def isNaN: Column = isnan
+
+  /**
    * True if the current expression is NaN.
    *
    * @group expr_ops
-   * @since 1.5.0
+   * @since 1.6.0
    */
-  def isNaN: Column = withExpr { IsNaN(expr) }
+  def isnan: Column = withExpr { IsNaN(expr) }
+
+  /**
+   * @group expr_ops
+   * @deprecated As of 1.6.0, replaced by `isnull`. This will be removed in Spark 2.0.
+   */
+  def isNull: Column = isnull
 
   /**
    * True if the current expression is null.
    *
    * @group expr_ops
-   * @since 1.3.0
+   * @since 1.6.0
    */
-  def isNull: Column = withExpr { IsNull(expr) }
+  def isnull: Column = withExpr { IsNull(expr) }
```

Comment:
I think its fine to be uniform with SQL in the functions that we call with columns as argument, but this seems like here we are breaking user code for no reason.  And if we are going to go lowercase then why are these `isnull` and not `is_null`?

For things that are just standard methods on objects I don't see a need to break with scala/java conventions.  We can have aliases in python to be compatible with pandas if that is the motivation.

/cc @rxin